docs/src/rules/no-underscore-dangle.md

```diff
@@ -3,16 +3,19 @@ title: no-underscore-dangle
 rule_type: suggestion
 ---
 
-
 As far as naming conventions for identifiers go, dangling underscores may be the most polarizing in JavaScript. Dangling underscores are underscores at either the beginning or end of an identifier, such as:
 
 ```js
 var _foo;
 ```
 
-There is actually a long history of using dangling underscores to indicate "private" members of objects in JavaScript (though JavaScript doesn't have truly private members, this convention served as a warning). This began with SpiderMonkey adding nonstandard methods such as `__defineGetter__()`. The intent with the underscores was to make it obvious that this method was special in some way. Since that time, using a single underscore prefix has become popular as a way to indicate "private" members of objects.
+There is a long history of marking "private" members with dangling underscores in JavaScript, beginning with SpiderMonkey adding nonstandard methods such as `__defineGetter__()`. Since that time, using a single underscore prefix has become the most popular convention for indicating a member is not part of the public interface of an object.
+
+::: important
+Official [Private class features](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields) were introduced in ECMAScript 2022. It is recommended you use those features when you need private members, rather than rely on conventions.
+:::
```

Comment:
We don't use asides like these in the intro material for rules, and a bit of cleanup.

```suggestion
Formal [private class fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields) were introduced in ECMAScript 2022. It is recommended you use private class fields when you need private members rather than relying on naming conventions.
```

Reply:
> Sure, noted for the future. I can't find any particular style guide for the docs anywhere, is this sort of thing just institutional memory?

Yeah, we don't have a style guide for docs. We tend to save asides for when they are really pointing out something that is important not to miss. In this case, it's just part of the normal flow of the text so there's really not a need for any extra callout.

> I was trying to keep consistent with the title on MDN

Ah, fair! I thought because the URL had "private class fields" that the title would, too.

> This feels a bit redundant 😄. I think They are recommended when you need private members instead of relying on naming conventions. covers being brief and hits the same points 😄

Yeah, it is a bit redundant, but "they" is unclear. For technical documentation, we always lean towards being redundant in order to improve clarity. (Words like "they", "it", etc., are best replaced with the actual things they are describing.)

I've updated the suggestion above.

Reply:
*tries not to bikeshed this too much*

If we rearrange slightly and combine into one sentence, I think that eliminates any ambiguity and gives us the brevity we're aiming at:

> We recommend using the formal [private class features](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields) introduced in ECMAScript 2022 for encapsulating private data and methods rather than relying on naming conventions.

Reply:
Works for me. Just make sure "recommend" -> "recommended"